system/include/emscripten/proxying.h

```diff
@@ -88,6 +89,32 @@ int emscripten_proxy_callback(em_proxying_queue* q,
                               void (*cancel)(void*),
                               void* arg);
 
+// Enqueue `func` on the given queue and thread. Once (and if) it finishes the
+// task by calling `emscripten_proxy_finish` on the given `em_proxying_ctx`, it
+// will asynchronously proxy `callback` back to the current thread on the same
+// queue, or if the target thread dies before the work can be completed,
+// `cancel` will be proxied back instead. All three function will receive the
+// same argument, `arg`. Returns 1 if `func` was successfully enqueued and the
+// target thread notified or 0 otherwise.
+int emscripten_proxy_callback_with_ctx(em_proxying_queue* q,
+                                       pthread_t target_thread,
+                                       void (*func)(em_proxying_ctx*, void*),
+                                       void (*callback)(void*),
+                                       void (*cancel)(void*),
+                                       void* arg);
+
+__attribute__((warn_unused_result)) em_promise_t
```

Comment:
What do you think about making an EM_WARN_UNUSED macro (in `emscripten/em_macros.h`) to make this a little less ugly?

Reply:
Sounds good. How about I do that as a follow up, since it would be good to update other uses of this attribute as well.

Reply:
SGTM